### PR TITLE
Add documentation for selective sync on Android

### PR DESCRIPTION
**Pull request by richardmcclellan**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/datastore/fragments/android/sync/50-selectiveSync.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+The max number of records that will be stored is configurable [here](~/lib/datastore/conflict.md).
+
+You can utilize selective sync to only persist a subset of your data instead.
+
+Selective sync works by applying predicates to the base and delta sync queries, as well as to incoming subscriptions.
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
```
Comment (jpignata) [inline docs/lib/datastore/fragments/android/sync/50-selectiveSync.md:13]: Is it worth throwing the `RxAmplify` variant here? I know it isn't much different, but for symmetry with the other examples.

File: docs/lib/datastore/fragments/android/sync/50-selectiveSync.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.addPlugin(AWSDataStorePlugin(DataStoreConfiguration.builder()
+    .syncExpression(Post::class.java) { Post.RATING.gt(5) }
+    .syncExpression(Comment::class.java) { Comment.STATUS.eq("active") }
+    .build()))
+Amplify.configure(context)
```
Comment (jpignata) [inline docs/lib/datastore/fragments/android/sync/50-selectiveSync.md:29]: Do we need to include the call to `configure`, you think?

**Review by jpignata — APPROVED** [review]
LGTM

File: docs/lib/datastore/fragments/android/sync/50-selectiveSync.md (hunk 2 above)
Comment (richardmcclellan) [reply docs/lib/datastore/fragments/android/sync/50-selectiveSync.md:29]: I just removed it in the latest commit - better to be brief I think!

File: docs/lib/datastore/fragments/android/sync/50-selectiveSync.md (hunk 1 above)
Comment (richardmcclellan) [reply docs/lib/datastore/fragments/android/sync/50-selectiveSync.md:13]: added in the latest commit!

**Pull request merged**